### ISSUE TITLE
Customize --dev options

### ISSUE THREAD

**Issue by Drup** [ocaml/dune#44]
I would like to be able to customize the options added by `--dev` (I personally dislike 6 and 9 in some code bases). 

I think it would also be nice to have `-keep-locs` and `-short-paths` there too.

**Comment by diml** [comment]
Indeed, I haven't really thought where this customization should go. It's probably something that should be configured per-project via a configuration file

**Comment by Drup** [comment]
Why not just in an arbitrary jbuilder, and it changes the default for all the folders that are under it ? You can still compose projects and it's fairly intuitive.

**Comment by diml** [comment]
True. I guess it could be something like this:

```scheme
(dev_mode
 ((flags (...))
  (ocamlc_flags (...))
  (ocamlopt_flags (...))))
```

In any case `-short-paths` looks fine by default. I've added it. I didn't know about `-keep-locs`, but I would imagine that this is more a global setting. Any reason you would want to enable it on a per project basis rather than globally? I imagine that you can do this with `OCAMLPARAM`

**Comment by Drup** [comment]
I use keep-locs because it improves the error messages for wrong signature inclusions a lot. It's pretty essential when you do weird things with functors. The trade-off is that it bloats `cmt*` files, so I don't need to use it for release.

**Comment by diml** [comment]
Ok, I tried it on jbuilder itself and the size difference of the `_build` directory was quite small, so I added `-keep-locs` to the dev mode by default.

That makes me think, in parsexp we have a [Positions module](https://github.com/janestreet/parsexp/blob/master/src/positions.mli) to encode a sequence of positions in a compact way. Maybe we could use something like this in the compiler to always keep locations.
